irohad/ametsuchi/impl/flat_file/flat_file.cpp

```diff
@@ -100,50 +116,24 @@ std::string FlatFile::directory() const {
 }
 
 Identifier FlatFile::last_id() const {
-  return current_id_.load();
+  return (available_blocks_.empty()) ? 0 : *available_blocks_.rbegin();
 }
 
 void FlatFile::dropAll() {
   iroha::remove_dir_contents(dump_dir_, log_);
-  auto res = FlatFile::check_consistency(dump_dir_, log_);
-  current_id_.store(*res);
+  available_blocks_.clear();
+}
+
+const BlockIdCollectionType &FlatFile::blockNumbers() const {
+  return available_blocks_;
 }
 
 // ----------| private API |----------
 
-FlatFile::FlatFile(Identifier current_id,
-                   const std::string &path,
+FlatFile::FlatFile(const std::string &path,
```

Comment:
Why not pass the path by value if we store its value?
```suggestion
FlatFile::FlatFile(std::string path,
```

Reply:
I see no reason to pass it by value. Here path will copied one time less at least.

Reply:
it is not important, but a general practice, if you need to store something inside a function, you should take it by value. I did not get why will it ever be copied when it was not, while you can move it. also you may leave it as is if you prefer.